src/models/metadata.ts

```diff
@@ -0,0 +1,63 @@
+import * as t from 'io-ts'
+import * as te from '../ancillary/io-ts-extra.js'
+
+/**
+ * Metadata fields gathered from genesis commit.
+ */
+export const GenesisFields = t.intersection(
```

Comment:
what would happen if in the future we add fields to the genesis header for some streamtype (in js-ceramic), but forget to add them here?

Reply:
These fields will be ignored.

Reply:
will that cause any behavior to be incorrect?  Like, will the fields actually be missing from the genesis structure we write to the database and use in the CAS?  If so, could that cause any problems, like making us recreate the wrong genesis commit CID anywhere?

Or is just that we don't get type checking on those fields, but they're still passed around in the background?

Reply:
These additional fields would be _completely_ ignored. We only use fields values in CAS, and only those fields mentioned in the code. We do not try to reconstruct CID or IPLD block from the fields for any purpose.

IMO, what should happen, is types and serde functions related to commits should be shared from `common` package.

Reply:
okay, thanks for clarifying.